### ISSUE TITLE
Alpha version seems to forget which kind of module system ember uses? 

### ISSUE THREAD

**Issue by NullVoxPopuli** [typed-ember/ember-cli-typescript#1163]
### Please paste the output of `ember -v` here
```
$ ember -v
ember-cli: 3.18.0
node: 14.4.0
os: linux x64
```

### Please paste the output of `tsc -v` here
```
Version 3.9.5
```

### Please paste the version of `ember-cli-typescript` and `ember-cli-typescript-blueprints` here

```
 yarn list --pattern 'ember-cli-typescript'
yarn list v1.22.4
warning package.json: "dependencies" has dependency "ember-cli-typescript" with range "4.0.0-alpha.1" that collides with a dependency in "devDependencies" of the same name with version "3.1.4"
warning Resolution field "ember-test-waiters@2.1.2" is incompatible with requested version "ember-test-waiters@^1.1.1"
├─ @glimmer/component@1.0.0
│  └─ ember-cli-typescript@3.0.0
├─ ember-app-scheduler@4.1.0
│  └─ ember-cli-typescript@2.0.2
├─ ember-cli-typescript-blueprints@3.0.0
├─ ember-cli-typescript@3.1.4
├─ ember-load-initializers@2.1.1
│  └─ ember-cli-typescript@2.0.2
└─ emberclear@0.0.1
   └─ ember-cli-typescript@4.0.0-alpha.1
Done in 2.45s.
```
however, on this PR / branch -- 
https://github.com/NullVoxPopuli/emberclear/pull/921/checks?check_run_id=746301255
one of my own deps is also the alpha release.

### Please paste your `tsconfig.json` and `tslint.json` or `eslint.json` (if applicable) below

<details><summary><b>My tsconfig.json</b></summary><pre>
{
  "compilerOptions": {
    "target": "ESNext",
    "allowJs": true,
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "noImplicitAny": true,
    "noImplicitThis": true,
    "alwaysStrict": true,
    "strictNullChecks": true,
    "strictPropertyInitialization": true,
    "noUnusedParameters": true,
    "noImplicitReturns": false,
    // unfortunately, noEmitOnError: true doesn't work well with linked TS addons :(
    // (or at least, linked JS addons with a TS declarations file....)
    "noEmitOnError": false,
    "noEmit": true,
    "inlineSourceMap": true,
    "inlineSources": true,
    "baseUrl": ".",
    "module": "esnext",
    "experimentalDecorators": true,
    "esModuleInterop": true,
    "noFallthroughCasesInSwitch": true,
    "paths": {
      "emberclear/tests/*": ["tests/*"],
      "emberclear/config/*": ["config/*"],
      "emberclear/*": ["app/*"],
      "@emberclear/ui/test-support/*": ["../addons/ui/addon-test-support/*"],
      "*": ["types/*", "../questionably-typed/*"]
    }
  },
  "include": [
    "app/**/*", "tests/**/*",
    "types/**/*",
    "../questionably-typed/**/*"
  ]
}

  
</pre></details>

<details><summary><b>My tslint.json or eslint.json</b></summary><pre>

// ;)

const {
  base,
  typescript,
  nodeOverrides,
  testOverrides,
  appOverrides,
} = require('@emberclear/config/eslint');

module.exports = {
  ...base,
  overrides: [
    // weird ones
    {
      files: ['app/services/prism-manager.ts'],
      ...typescript,
      rules: {
        ...typescript.rules,
        'no-undef': 'off',
      },
    },
    {
      ...appOverrides,
      parserOptions: {
        ...base.parserOptions,
        project: require.resolve('./tsconfig.json'),
      },
    },
    testOverrides,
    nodeOverrides,
  ],
};

</pre></details>

### What are instructions we can follow to reproduce the issue?
```sh
# clone https://github.com/NullVoxPopuli/emberclear/pull/921
git clone git@github.com:NullVoxPopuli/emberclear.git
cd emberclear
git checkout alpha-ec-ts
cd client/web/emberclear
yarn
yarn ember test
```

##### Reproduction Case

Base: https://codesandbox.io/s/github/NullVoxPopuli/ember-new-output/tree/stable-ts
 -  Based off Base: https://codesandbox.io/s/github/NullVoxPopuli/ember-new-output/tree/stable
    - `ember install ember-cli-typescript`

Repro: https://codesandbox.io/s/zen-currying-1ufwr
  - upgraded ember-cli-typescript to the alpha
  - added a package to be used by ember-auto-import
  - import anything from that package (foo.ts)

### Now about that bug. What did you expect to see?
No error :)

### What happened instead?
This error: https://github.com/NullVoxPopuli/emberclear/pull/921/checks?check_run_id=746301255

```
 TypeError: Cannot read property 'exports' of undefined
                at Module._reify (https://0.0.0.0:7357/assets/vendor.js:155:59)
                at Module.reify (https://0.0.0.0:7357/assets/vendor.js:142:27)
                at Module.exports (https://0.0.0.0:7357/assets/vendor.js:116:10)
                at Module._reify (https://0.0.0.0:7357/assets/vendor.js:155:59)
                at Module.reify (https://0.0.0.0:7357/assets/vendor.js:142:27)
                at Module.exports (https://0.0.0.0:7357/assets/vendor.js:116:10)
                at requireModule (https://0.0.0.0:7357/assets/vendor.js:39:18)
                at TestLoader.require (https://0.0.0.0:7357/assets/test-support.js:46698:9)
                at TestLoader.loadModules (https://0.0.0.0:7357/assets/test-support.js:46691:14)
                at loadTests (https://0.0.0.0:7357/assets/test-support.js:49709:22)
```

**Comment by NullVoxPopuli** [comment]
it looks like this is more related to babel than ec-ts, which is good news!

https://github.com/NullVoxPopuli/emberclear/pull/927/commits/291af7b9dcc88b93fa32a2e883cbe3a7fe8c7595

this commit removes ember-cli-typescript and adds the required babel configs that ec-ts would otherwise provide for me.

I'm sure other people will run in to this, so I'm going to be documenting my debugging adventures here :D

I'm starting with deleting as many tests as I can, running ember test between deletions to see if I can get a quickly repeatable failure case, and then start deleting app code from there -- after that, I'll start removing packages from the package.json and see what happens.

**Comment by NullVoxPopuli** [comment]
When I eliminated all but a couple tests, I noticed an import error. It couldn't find the package "qrcode", which was handled by ember-auto-import.

After deleting the import from qrcode, and commenting out the code that uses it, my build / tests passed.

Note that qrcode is found and does what it's supposed to do pre-babel-only config and pre-ec-ts alpha

**Comment by NullVoxPopuli** [comment]
I've restored all my deleted files, and then deleted all my unit and integration tests, leaving the acceptance tests folder.

With less output, I'm able to see that between the 
```
            TypeError: Cannot read property 'exports' of undefined
```
errors, there are a number of ember-auto-import imported modules that "can't be found".

cc @ef4 (just for visibility, not asking for you to solve, I'm still investigating)

**Comment by NullVoxPopuli** [comment]
I can confirm that removing all imports that ember-auto-import would handle "fixes" the 
```
TypeError: Cannot read property 'exports' of undefined
```
error.

**Comment by NullVoxPopuli** [comment]
Same issue with Embroider :thinking:

**Comment by NullVoxPopuli** [comment]
ah ha! A repro! https://codesandbox.io/s/zen-currying-1ufwr?file=/app/templates/application.hbs

**Comment by chriskrycho** [comment]
@NullVoxPopuli can you summarize the actual issue here? Is there anything *actually* at issue with e-c-ts 4?

**Comment by NullVoxPopuli** [comment]
Last I looked, it actually wasn't an ects issue, but an ember-auto-import one.
